src/qibo/imaging/mpldrawer.py

```diff
@@ -0,0 +1,481 @@
+                return r"$|%s\rangle$" % inits[label]
+        return r"$|%s\rangle$" % label
+
+    def _make_cluster_gates(self, gates_items):
+        cluster_gates = []
+        temp_gates = []
+        temp_mgates = []
+        for i in list(range(len(gates_items))):
+            item = gates_items[i]
+
+            if (len(item) == 2) and i > 0 and "MEASURE" not in item[0]:
+                if len(temp_gates) > 0 and item[1] == gates_items[i - 1][1]:
+                    gates = []
+                temp_gates.append(item)
+            elif "MEASURE" in item[0]:
+                temp_mgates.append(item)
+            else:
+                if len(temp_gates) != 0:
+                    cluster_gates.append(temp_gates)
+                    temp_gates = []
+
+                if len(temp_mgates) != 0:
+                    cluster_gates.append(temp_mgates)
+                    temp_mgates = []
+
+                if "MEASURE" not in item[0]:
+                    cluster_gates.append([item])
+            i = i + 1
+
+        if len(temp_gates) > 0:
+            cluster_gates.append(temp_gates)
+
+        if len(temp_mgates) > 0:
+            cluster_gates.append(temp_mgates)
+
+        temp_gates = []
+        temp_mgates = []
+
+        return cluster_gates
+
+    def plot_qibo_circuit(self, circuit, scale, cluster_gates):
+
+        inits = list(range(circuit.nqubits))
+
+        labels = []
+        for i in range(circuit.nqubits):
+            labels.append("q_" + str(i))
+
+        if len(circuit.queue) > 0:
+            gates_plot = []
+
+            for gate in circuit.queue:
+                init_label = gate.name.upper()
+
+                if len(gate._control_qubits) > 0 and "C" in init_label[0]:
+                    init_label = init_label[1:]
+
+                item = ()
+                item += (init_label,)
+
+                for qbit in gate._target_qubits:
+                    if qbit is tuple:
+                        item += ("q_" + str(qbit[0]),)
+                    else:
+                        item += ("q_" + str(qbit),)
+
+                for qbit in gate._control_qubits:
+                    if qbit is tuple:
+                        item += ("q_" + str(qbit[0]),)
+                    else:
+                        item += ("q_" + str(qbit),)
+
+                gates_plot.append(item)
+
+            if cluster_gates:
+                gates_cluster = self._make_cluster_gates(gates_plot)
+                return self._plot_quantum_schedule(
+                    gates_cluster, inits, labels, scale=scale
+                )
+
+            return self._plot_quantum_circuit(gates_plot, inits, labels, scale=scale)
+        else:
+            return self._plot_lines_circuit(inits, labels, scale=scale)
+
+    @staticmethod
+    def save_fig(fig, path_file):
+        """Save matpltlib image to file
+        path_file   Path to file with filename and extension
+        """
+        return fig.savefig(path_file, bbox_inches="tight")
```

Comment:
This function is just calling a function, and it is invoked only once.

Just drop it and replace its invocation with a direct call to `fig.savefig()`.

Reply:
I removed this function by a direct call to `savefig`